src/tests/system/tests/test_ipa.py

```diff
@@ -390,3 +390,122 @@ def test_ipa__check_gssapi_authentication_indicator(client: Client, ipa: IPA):
+        3. Check IPA users certificate
+    :expectedresults:
+        1. Successfully get the user's overridden attributes'
+        2. Successfully get the group's overridden attributes'
+        3. Sccessfully get the user's certificate
+    :customerscenario: False
+    """
+    ipa.idview("testview1").add(description="This is a new view")
+    ipa.idview("testview1").apply(hosts=[f"{client.host.hostname}"])
+
+    with open(f"{moduledatadir}/certificate") as f:
+        certificate_content = f.read().strip()
+
+    ipa.user("user-1").add().iduseroverride().add_override(
+        "testview1",
+        uid=1234567,
+        gid=7654321,
+        gecos="This is the ID user override",
+        home_directory="/home/newhomedir",
+        shell="/bin/newloginshell",
+        login="newuser",
+        certificate=certificate_content,
+    )
+    ipa.group("group-1").add().idgroupoverride().add_override(
+        idview_name="testview1", group_name="newgroup", gid=88888, description="This is ID group override"
+    )
+
+    client.sssd.restart()
+
+    result = ipa.user("user-1").iduseroverride().show_override("newview1")
+
+    for user in ["user-1", "newuser"]:
+        userlookup = client.tools.getent.passwd(user)
+        assert userlookup.uid == 1234567, "overriden uid not found"
+        assert userlookup.name == "newuser", "overriden login name not found"
+        assert userlookup.gid == 7654321, "overriden gid not found"
+        assert userlookup.home == "/home/newhomedir", "overriden home directory not found"
+        assert userlookup.shell == "/bin/newloginshell", "overriden shell not found"
+
+    for group in ["group-1", "newgroup"]:
+        grouplookup = client.tools.getent.group(group)
+        assert grouplookup.name == "newgroup", "overriden group name not found"
+        assert grouplookup.gid == 88888, "overriden gid not found"
+
+    assert certificate_content in result.get("usercertificate", [""])[0], "Certificate content mismatch!"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__idview_non_override_attribute(client: Client, ipa: IPA):
+    """
+    :title: Verify an IPA ID view can not override non override attributes on the client
+    :description:
+        To ensure that an IPA ID view can not override non override attributes on the client system.
+    :setup:
+        1. Client is configured to use ID view
+        2. Add a user with non override attributes
+        4. Restart SSSD on client
+    :steps:
+        1. Check IPA user's non override attributes should not be overridden
+    :expectedresults:
+        1. Non override attributes should not be overridden
+    :customerscenario: False
+    """
+
+    ipa.idview("testview").add(description="This is a new view")
+    ipa.idview("testview").apply(hosts=[f"{client.host.hostname}"])
+
+    ipa.user("user-1").add().iduseroverride().add_override("testview", email="email.com", fname="fname", lname="lname")
```

Comment:
I don't quite understand this test. What happens when providing `email`,  `fname`, and `lname` here? Shouldn't this `add_override` method fail as these are not valid user ID overrides ?

Reply:
It won't fail but skip the not valid attribute, https://github.com/SSSD/sssd-test-framework/pull/154/files#diff-a97a14df2c8e3f1d6e9c0878a73fed25bbbe15074d0ad17f47f82d4f14b4f259R759
I want your suggestion here as what should I keep,
1. It will skip the attribute if those are not-valid and only create ID override (currently I have added this). 
2. It will raise stderr when we add with non-valid attribute.

Reply:
In my opinion option 2 makes more sense, error out if attempting to use 'add_override' for an attribute that is not a valid override.

Reply:
I updated the code, please do check again.

Reply:
Thank you, that looks better but now this test is only actually testing the `add_override()` ID user override method. In this case I don't think we really need this test, it is not actually testing any SSSD functionality. What do you think?